### ISSUE TITLE
Bug: [v8.0.0-alpha.10] incorrect error message `Parsing error: "parserOptions.programs"`

### ISSUE THREAD

**Issue by OlivierZal** [typescript-eslint/typescript-eslint#9096]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

_Bug discussed and confirmed by @JoshuaKGoldberg and @bradzacher._

Config working with `typescript-eslint` v7.9.0 does not work with v8.0.0-alpha.10, raising:
```
/Users/<...>/eslint.config.js
  0:0  error  Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): eslint.config.js
```
although I don't use any home-made program.

Potential culprit: https://github.com/typescript-eslint/typescript-eslint/pull/8834/files

### Reproduction Repository Link

https://github.com/OlivierZal/melcloud-api/blob/e118342a2ce16c49e41e145844bc318e344cfe52/eslint.config.js

### Repro Steps

1) `git clone https://github.com/OlivierZal/melcloud-api`
2) `git co e118342a2ce16c49e41e145844bc318e344cfe52`
3) `eslint .`
4) `npm i typescript-eslint@8.0.0-alpha.10 --force`
5) `eslint .`
6) compare the 2 outputs (see attached screenshots: v7.9.0 vs v8.0.0-alpha)

![v7](https://github.com/typescript-eslint/typescript-eslint/assets/88216225/ec34f899-569f-4b29-9566-488acf22f3a0)
![v8](https://github.com/typescript-eslint/typescript-eslint/assets/88216225/d6924411-191c-41ca-bc1c-9c9fdf5af976)

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.0.0-alpha.10` |
| `TypeScript`                           | `5.4.5` |
| `ESLint`                               | `9.2.0` |
| `node`                                 | `22.1.0` |

**Comment by bradzacher** [comment]
Actually I think the error **message** is wrong but the error itself is correct.

You have type-aware linting turned on:
https://github.com/OlivierZal/melcloud-api/blob/e118342a2ce16c49e41e145844bc318e344cfe52/eslint.config.js#L15

Then you are using `...tsEslint.configs.disableTypeChecked.rules` to disable the type-aware rules
https://github.com/OlivierZal/melcloud-api/blob/e118342a2ce16c49e41e145844bc318e344cfe52/eslint.config.js#L244

However this is a misuse of the config because the config has more than just `.rules` - it also 
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages%2Feslint-plugin%2Fsrc%2Fconfigs%2Fdisable-type-checked.ts#L11 

So because you're misusing the config - you never disable type-aware _parsing_ for JS files. You've disabled all the rules but the parser is still generating types for JS files. Hence the parser is erroring on a JS file because it's not included in a tsconfig.

Docs on correct use of the config
https://typescript-eslint.io/users/configs#disable-type-checked
Or alternately (as mentioned on discord) using the `extends` prop provided by our config util:
https://typescript-eslint.io/packages/typescript-eslint#flat-config-extends

----

So yeah - the message is wrong - it should say `Parsing error: "parserOptions.projects" has been provided ...`
But the error itself is correct.
